typescript/cli/src/commands/hook.ts

```diff
@@ -0,0 +1,54 @@
+import { CommandModule } from 'yargs';
+
+import { readHookConfig } from '../hook/read.js';
+import { log } from '../logger.js';
+
+import {
+  addressCommandOption,
+  chainCommandOption,
+  chainsCommandOption,
+  concurrencyCommandOption,
+  fileFormatOption,
+  outputFileOption,
+} from './options.js';
+
+/**
+ * Parent command
+ */
+export const hookCommand: CommandModule = {
+  command: 'hook',
+  describe: 'Operations relating to Hooks',
+  builder: (yargs) => yargs.command(read).version(false).demandCommand(),
+  handler: () => log('Command required'),
+};
+
+export const read: CommandModule = {
+  command: 'read',
+  describe: 'Reads onchain Hook configuration for a given address',
+  builder: (yargs) =>
+    yargs.options({
+      chains: chainsCommandOption,
+      chain: {
+        ...chainCommandOption,
+        demandOption: true,
+      },
+      address: {
+        ...addressCommandOption,
+        demandOption: true,
+      },
+      concurrency: concurrencyCommandOption,
+      format: fileFormatOption,
+      output: outputFileOption(),
```

Comment:
can we make these global somehow?

Reply:
also would expect default to be `stdout`

Reply:
We can but I'm already reworking the common args so let's leave it as is